contributingGuides/TS_STYLE.md

```diff
@@ -0,0 +1,403 @@
+# Expensify TypeScript Style Guide
+
+## Table of Contents
+
+- [Other Expensify Resources on TypeScript](#other-expensify-resources-on-typescript)
+- [Learning Resources](#learning-resources)
+- [Exception to Rules](#exception-to-rules)
+- [1.1 Naming Conventions](#convension-naming-convension)
```

Comment:
NAB: these links are inconsistent "convension" vs "convensions" vs "convensions" - that may lead to mistakes when linking

Reply:
still editing details 😓  but I appreciate your detailed reviews 🙇  let's focus on high level reviews until this PR is marked as ready 👍